xbmc/settings/dialogs/GUIDialogSettingsBase.cpp

```diff
@@ -599,7 +599,7 @@ std::set<std::string> CGUIDialogSettingsBase::CreateSettings()
     for (SettingList::const_iterator settingIt = settings.begin(); settingIt != settings.end();
          ++settingIt)
     {
-      std::shared_ptr<CSetting> pSetting = *settingIt;
+      const std::shared_ptr<CSetting>& pSetting = *settingIt;
```

Comment:
I actually fixed a bug doing the reverse of this, but it should be fine here in `CGUIDialogSettingsBase::CreateSettings()`